cirq/ops/pauli_string.py

```diff
@@ -163,27 +164,52 @@ def _value_equality_values_cls_(self):
             return gate_operation.GateOperation
         return PauliString
 
-    def equal_up_to_coefficient(self, other: 'PauliString') -> bool:
+    def equal_up_to_coefficient(self, other: 'cirq.PauliString') -> bool:
         return self._qubit_pauli_map == other._qubit_pauli_map
 
-    def __getitem__(self, key: 'cirq.Qid') -> pauli_gates.Pauli:
+    def __getitem__(self, key: TKey) -> pauli_gates.Pauli:
         return self._qubit_pauli_map[key]
 
     # pylint: disable=function-redefined
     @overload
-    def get(self, key: 'cirq.Qid', default: None = None) -> pauli_gates.Pauli:
+    def get(self, key: Any, default: None = None) -> pauli_gates.Pauli:
         pass
 
     @overload
-    def get(self, key: 'cirq.Qid',
+    def get(self, key: Any,
             default: TDefault) -> Union[pauli_gates.Pauli, TDefault]:
         pass
 
-    def get(self, key: 'cirq.Qid', default=None):
+    def get(self, key: Any, default=None):
         return self._qubit_pauli_map.get(key, default)
-    # pylint: enable=function-redefined
 
-    def __mul__(self, other) -> 'PauliString':
+    @overload
+    def __mul__(  # type: ignore
+            self, other: 'cirq.PauliString[TKey2]'
+    ) -> 'cirq.PauliString[Union[TKey, TKey2]]':
+        pass
+
+    @overload
+    def __mul__(self, other: Mapping[TKey2, 'cirq.PAULI_GATE_LIKE']
+               ) -> 'cirq.PauliString[Union[TKey, TKey2]]':
+        pass
+
+    @overload
+    def __mul__(self, other: Iterable['cirq.PAULI_STRING_LIKE']
+               ) -> 'cirq.PauliString[Union[TKey, cirq.Qid]]':
+        pass
+
+    @overload
+    def __mul__(self, other: 'cirq.Operation'
+               ) -> 'cirq.PauliString[Union[TKey, cirq.Qid]]':
```

Comment:
why use cirq.Qid here directly? Wasn't that the point with TKeyX?

Reply:
The issue is that `other` is an operation over qubits, and is not required to be over the same type of qubits as the pauli string. We would need to make Operation generic to guarantee keeping `TKey`, which I do not think is viable since Operation is inherited by so many things.

Reply:
Got it, makes sense.